### ISSUE TITLE
data-prep script should be updated to support streaming

### ISSUE THREAD

**Issue by AmirAlavi** [AmirAlavi/scrna_nn#21]
Note #20, this is related. As of now, I think the best plan of action is to continue to write pandas DataFrames as HDF5 binary files, but making them streaming-friendly.
I think this is possible: pandas DataFrames can be written to HDF5 via pandas.DataFrame.to_hdf(..., format='table'), and according to documentation (https://pandas.pydata.org/pandas-docs/stable/generated/pandas.DataFrame.to_hdf.html) this will allow for subseting.
Now, in keras, there is HDF5Matrix (https://keras.io/utils/) which seems to allow subsetting and even normalization.

This means that now the data-prep script should also compute whole-dataset statistics (gene mean, gene std, gene min, gene max) and save them, in case they are needed for normalization downstream (otherwise, we'd have to load the entire dataset to compute these statistics later, defeating the whole purpose of streaming).
